v1/plugin/stream_collector_proxy.go

```diff
@@ -1,119 +1,179 @@
 package plugin
 
 import (
+	"errors"
 	"fmt"
+	"time"
 
 	"golang.org/x/net/context"
 
 	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin/rpc"
 )
 
+const (
+	defaultMaxCollectDuration = 10 * time.Second
+	defaultMaxMetricsBuffer   = 0
+)
+
 type StreamProxy struct {
 	pluginProxy
-
 	plugin StreamCollector
+
+	// maxMetricsBuffer is the maximum number of metrics the plugin is buffering before sending metrics.
+	// Defaults to zero what means send metrics immediately.
+	maxMetricsBuffer int64
+
+	// maxCollectionDuration sets the maximum duration (always greater than 0s) between collections before metrics are sent.
+	// Defaults to 10s what means that after 10 seconds no new metrics are received, the plugin should send
+	// whatever data it has in the buffer instead of waiting longer.
+	maxCollectDuration time.Duration
+}
+
+func (p *StreamProxy) GetMetricTypes(ctx context.Context, arg *rpc.GetMetricTypesArg) (*rpc.MetricsReply, error) {
```

Comment:
**To reviewers**
There is no change in `GetMetricsTypes`. I just moved that function to the top as it is exported - see [here](https://github.com/intelsdi-x/snap-plugin-lib-go/pull/77/files#diff-aa6fc80eb3b8c2400af503a0443c72b7L101)